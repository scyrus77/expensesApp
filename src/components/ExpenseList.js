import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.map(expense => (<ExpenseListItem  key={ expense.id } { ...expense } />))
    }
  </div>
);

const mapStateToProps = ({ expenses, filters }) => {
  return {
    expenses: selectExpenses(expenses, filters)
  }
};

export default connect(mapStateToProps)(ExpenseList);

