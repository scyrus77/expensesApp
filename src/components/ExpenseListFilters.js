import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';

import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends Component {

  state = {
    calendarFocused: null,
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState( () => ({calendarFocused}) );
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    // console.log('Select onChange', e.target.value);
    if (e.target.value === 'amount') {
      this.props.sortByAmount();
    } else if (e.target.value === 'date') {
      this.props.sortByDate();
    }
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={ this.props.filters.text }
          onChange={ this.onTextChange }
        />
        <select
          value={ this.props.filters.sortBy }
          onChange={ this.onSortChange }
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>

        <DateRangePicker
          startDate={ this.props.filters.startDate }
          startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
          endDate={ this.props.filters.endDate } // momentPropTypes.momentObj or null,
          endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
          onDatesChange={ this.onDatesChange }
          focusedInput={ this.state.calendarFocused }
          onFocusChange={ this.onFocusChange }
          showClearDates={ true }
          numberOfMonths={ 1 }
          isOutsideRange={ () => false }
        />

      </div>
    )
  }

}

const mapStateToProps = ({ filters }) => ({ filters });

const mapDispatchProps = (dispatch) => ({
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate()),
});

export default connect(mapStateToProps, mapDispatchProps)(ExpenseListFilters);
