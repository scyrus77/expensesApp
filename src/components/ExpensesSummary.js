import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = (props) => (
  <div>
    <h1>Expenses Summary</h1>
    <div>
      Viewing { props.expenses.length }
      { props.expenses.length > 1 ? ' expenses ' : ' expense ' }
      totalling { numeral(selectExpensesTotal(props.expenses)).format('$0,0.00') } </div>
  </div>
);

const mapStateToProps = ({ expenses, filters }) => {
  return {
    expenses: selectExpenses(expenses, filters)
  }
};

export default connect(mapStateToProps)(ExpensesSummary);


