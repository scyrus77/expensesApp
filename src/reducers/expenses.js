const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  // console.log('expenseReducer', action);
  switch (action.type) {
    case 'ADD_EXPENSE':
      // console.log('ADD_EXPENSE', action);
      return [...state, action.expense];
    case 'EDIT_EXPENSE':
      // console.log('EDIT_EXPENSE', action);
      return state.map( expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          }
        } else {
          return expense;
        }
      });
    case 'REMOVE_EXPENSE':
      // console.log('REMOVE_EXPENSE', action);
      return state.filter(({ id }) => id !== action.id );
    case 'SET_EXPENSES':
      // console.log('SET_EXPENSES', action);
      return action.expenses;
    default:
      return state;
  }

};
