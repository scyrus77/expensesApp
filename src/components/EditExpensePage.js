import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      <h1>Edit Expense</h1>
      <ExpenseForm
        expense={ props.expense }
        onSubmit={(expense) => {
          console.log('EditExpensePage', props, expense);
          props.dispatch(editExpense(props.expense.id, expense));
          props.history.push('/');
        }}
        onDelete={() => {
          console.log('EditExpensePage onDelete');
          props.dispatch(removeExpense({ id: props.expense.id }));
          props.history.push('/');
        }}
      />
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find( expense => expense.id === props.match.params.id )
  }
};

export default connect(mapStateToProps)(EditExpensePage);

