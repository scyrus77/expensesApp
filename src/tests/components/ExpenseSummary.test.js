import React from 'react'
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={ expenses } />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={ [] } />);
  expect(wrapper).toMatchSnapshot();
});

