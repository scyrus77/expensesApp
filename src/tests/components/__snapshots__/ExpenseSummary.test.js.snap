// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with expenses 1`] = `
<div>
  <h1>
    Expenses Summary
  </h1>
  <div>
    Viewing 
    3
     expenses 
    totalling 
    $114,195.00
     
  </div>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `
<div>
  <h1>
    Expenses Summary
  </h1>
  <div>
    Viewing 
    0
     expense 
    totalling 
    $0.00
     
  </div>
</div>
`;
